source/extensions/filters/http/ext_authz/ext_authz.h

```diff
@@ -94,6 +96,24 @@ class FilterConfig {
         destination_labels_[labels_it.first] = labels_it.second.string_value();
       }
     }
+
+    if (config.has_allowed_headers() &&
+        config.http_service().authorization_request().has_allowed_headers()) {
+      ExceptionUtil::throwEnvoyException("Invalid duplicate configuration for allowed_headers.");
+    }
+
+    if (config.has_grpc_service() && config.has_allowed_headers()) {
+      request_header_matchers_ = Filters::Common::ExtAuthz::CheckRequestUtils::toRequestMatchers(
```

Comment:
Please add comment that unset request_header_matchers_ means to allow all, and that this is done for backwards compatibility.